gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageItemInfo.java

```diff
@@ -337,6 +337,15 @@ public boolean isDirectory() {
     return isGlobalRoot() || isBucket() || resourceId.isDirectory();
   }
 
+  /** Indicates whether {@code itemInfo} is an inferred directory */
+  public boolean isInferredDirectory() {
+    return creationTime == 0
```

Comment:
May you search codebase if we are doing this check anywhere else, and use this method instead if there are such places?

Reply:
Did not find any methods that mimic this behaviour and i believe this method belongs to this class so that the definition of inferredDirectory in contained in the same class where it is created (via the factory method)